File: python/problem4.py
# problem4.py
def par_o_impar(n: int) -> str:
    """
    Determina si un numero entero es par o impar.
    """
    return "Par" if n % 2 == 0 else "Impar"


if __name__ == "__main__":
    try:
        numero = int(input("Ingresa un numero entero: "))
        print(f"El numero {numero} es {par_o_impar(numero)}")
    except ValueError:
        print("Error: debes ingresar un numero entero valido.")

File: haskell/problem4.hs
-- Problem4.hs
module Problem4 (parOImpar) where

-- Funcion que determina si un numero entero es par o impar
parOImpar :: Int -> String
parOImpar n
    | even n    = "Par"
    | otherwise = "Impar"

-- funcion principal de prueba
main :: IO ()
main = do
    print (parOImpar 4)    -- "Par"
    print (parOImpar 7)    -- "Impar"
    print (parOImpar 0)    -- "Par"
    print (parOImpar (-3)) -- "Impar"
